packages/accounts/src/msca/plugins/multisig/signer.ts

```diff
@@ -0,0 +1,104 @@
+import type {
+  Address,
+  BundlerClient,
+  SmartAccountSigner,
+} from "@alchemy/aa-core";
+import {
+  hashMessage,
+  hashTypedData,
+  type Hash,
+  type Hex,
+  type SignableMessage,
+  type Transport,
+  type TypedData,
+  type TypedDataDefinition,
+} from "viem";
+import { MultisigPlugin, MultisigPluginAbi } from "./plugin.js";
+
+type MultisigSignMethodsParams<
+  TTransport extends Transport,
+  TSigner extends SmartAccountSigner
+> = {
+  client: BundlerClient<TTransport>;
+  accountAddress: Address;
+  signer: () => TSigner;
+  threshold: bigint;
+  pluginAddress?: Address;
+};
+
+export const multisigSignMethods = <
+  TTransport extends Transport,
+  TSigner extends SmartAccountSigner
+>({
+  client,
+  accountAddress,
+  signer,
+  threshold,
+  pluginAddress = MultisigPlugin.meta.addresses[client.chain.id],
+}: MultisigSignMethodsParams<TTransport, TSigner>) => {
+  const signWith712Wrapper = async (msg: Hash): Promise<`0x${string}`> => {
+    const [, name, version, chainId, verifyingContract, salt] =
+      await client.readContract({
+        abi: MultisigPluginAbi,
+        address: pluginAddress,
+        functionName: "eip712Domain",
+        account: accountAddress,
+      });
+
+    return signer().signTypedData({
+      domain: {
+        chainId: Number(chainId),
+        name,
+        salt,
+        verifyingContract,
+        version,
+      },
+      types: {
+        AlchemyMultisigMessage: [{ name: "message", type: "bytes" }],
+      },
+      message: {
+        message: msg,
+      },
+      primaryType: "AlchemyMultisigMessage",
+    });
+  };
+
+  return {
+    getDummySignature: async (): Promise<`0x${string}`> => {
```

Comment:
should this be Promise<Hex> instead?

Reply:
hex is just a type alias for `0x${string}` so its fine

Reply:
we can change if you want

Reply:
TIL